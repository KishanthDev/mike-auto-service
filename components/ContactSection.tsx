"use client";

import Image from "next/image";
import React from "react";

export default function ContactSection() {
  return (
    <section>
      <h2 className="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white">
        Contact & Location
      </h2>

      <div className="mt-4 md:mt-6 space-y-3 text-sm md:text-base text-gray-700 dark:text-gray-300">
        <p>
          <strong>Address:</strong>
          <br />
          1234 Auto Drive
          <br />
          Springfield, IL 62701
        </p>

        <div className="w-full h-[200px] md:h-[250px] bg-gray-200 dark:bg-gray-700 rounded mt-3 overflow-hidden">
          <Image
            src="/api/placeholder/400/250"
            alt="Map"
            width={400}
            height={250}
            className="w-full h-full object-cover"
          />
        </div>

        <p className="mt-3">
          <strong>Phone:</strong> (555) 123-4567
        </p>

        <p>
          <strong>Email:</strong> service@mikesauto.com
        </p>

        <p>
          <strong>Website:</strong>{" "}
          <a
            href="https://www.mikesauto.com"
            className="text-blue-600 dark:text-blue-400 hover:underline"
            target="_blank"
            rel="noopener noreferrer"
          >
            www.mikesauto.com
          </a>
        </p>
      </div>
    </section>
  );
}
