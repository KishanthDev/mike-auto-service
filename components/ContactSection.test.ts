import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ContactSection from "./ContactSection";

// Mock the next/image component
jest.mock("next/image", () => {
    const MockedImage = ({ src, alt, width, height, className }: any) => {
      return React.createElement("img", {
        src,
        alt,
        width,
        height,
        className,
        "data-testid": "mock-image",
      });
    };
    MockedImage.displayName = "MockedNextImage"; // ðŸ‘ˆ Fixes the error
    return MockedImage;
  });
  

describe("ContactSection", () => {
  it("renders the section with correct heading", () => {
    render(React.createElement(ContactSection));

    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Contact & Location");
    expect(heading).toHaveClass(
      "text-2xl",
      "md:text-3xl",
      "font-bold",
      "border-b-2",
      "border-red-600",
      "pb-2",
      "text-[#1a3c6e]",
      "dark:text-white",
    );
  });

  it("renders the address information correctly", () => {
    render(React.createElement(ContactSection));

    const addressLabel = screen.getByText("Address:");
    expect(addressLabel).toBeInTheDocument();
    expect(addressLabel.tagName).toBe("STRONG");
  });

  it("renders the map image container", () => {
    render(React.createElement(ContactSection));

    const mapContainer = screen.getByTestId("mock-image").parentElement;
    expect(mapContainer).toHaveClass(
      "w-full",
      "h-[200px]",
      "md:h-[250px]",
      "bg-gray-200",
      "dark:bg-gray-700",
      "rounded",
      "mt-3",
      "overflow-hidden",
    );

    const image = screen.getByTestId("mock-image");
    expect(image).toHaveAttribute("src", "/api/placeholder/400/250");
    expect(image).toHaveAttribute("alt", "Map");
    expect(image).toHaveClass("w-full", "h-full", "object-cover");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(React.createElement(ContactSection));
    expect(asFragment()).toMatchSnapshot();
  });
});
