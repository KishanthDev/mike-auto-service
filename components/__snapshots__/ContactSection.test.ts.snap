// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactSection matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Contact & Location
    </h2>
    <div
      class="mt-4 md:mt-6 space-y-3 text-sm md:text-base text-gray-700 dark:text-gray-300"
    >
      <p>
        <strong>
          Address:
        </strong>
        <br />
        1234 Auto Drive
        <br />
        Springfield, IL 62701
      </p>
      <div
        class="w-full h-[200px] md:h-[250px] bg-gray-200 dark:bg-gray-700 rounded mt-3 overflow-hidden"
      >
        <img
          alt="Map"
          class="w-full h-full object-cover"
          data-testid="mock-image"
          height="250"
          src="/api/placeholder/400/250"
          width="400"
        />
      </div>
      <p
        class="mt-3"
      >
        <strong>
          Phone:
        </strong>
         (555) 123-4567
      </p>
      <p>
        <strong>
          Email:
        </strong>
         service@mikesauto.com
      </p>
      <p>
        <strong>
          Website:
        </strong>
         
        <a
          class="text-blue-600 dark:text-blue-400 hover:underline"
          href="https://www.mikesauto.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          www.mikesauto.com
        </a>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
